applications/ShapeOptimizationApplication/custom_utilities/mapping/mapper_vertex_morphing.h

```diff
@@ -468,13 +468,16 @@ class MapperVertexMorphing : public Mapper
     void InitializeMappingVariables()
     {
         const unsigned int origin_node_number = mrOriginModelPart.Nodes().size();
-        const unsigned int destination_node_number = mrDestinationModelPart.Nodes().size();
-
-        mMappingMatrix.resize(destination_node_number,origin_node_number,false);
-        mMappingMatrix.clear();
+        mValuesOrigin.resize(3);
+        mValuesOrigin[0] = ZeroVector(origin_node_number);
```

Comment:
@sunethwarna did this solve the issue?

Reply:
The problem having 3rd component in 2D was solved by this. Earlier, when this is called it only resizes the vectors to the initial configuration only because on the next iterations it ignores because the `std::vector` is already correctly sized, but the `Vector` was not sized properly, and it was reading garbage values. So this was solved, now I don't have a problem with the mapped sensitivities.

Now I have a problem in the `MeshMotionSolver`. So little by little we are getting there :)